### PR TITLE
Stop excluding non-current listed versions in code search

### LINKED ISSUE #13431: Stop excluding non-current listed versions in code search
For the initial implementation of Scanner query rule task, we restricted the versions it runs on, excluding non-current listed versions. Now that we have done some initial testing, we've determined we no longer want to restrict versions that much.

STR:
- Submit a listed add-on, and then submit a new version for that add-on.
- Create a Scanner Query Rule that would match something in both versions and run it

Actual Results:
- Only the currently public version is in the results

Expected Results
- The 2 versions are found

### PR DESCRIPTION
**Pull request by diox**
Fixes #13431

### CODE REVIEW

File: src/olympia/scanners/tasks.py [hunk 1]
```diff
@@ -238,8 +237,10 @@ def mark_yara_query_rule_as_completed_or_aborted(query_rule_pk):
     rule = ScannerQueryRule.objects.get(pk=query_rule_pk)
     try:
         if rule.state == RUNNING:
+            log.info('Marking Yara Query Rule %s as completed', rule.pk)
```
Comment (eviljeff) [inline src/olympia/scanners/tasks.py]: these two `log.info` statements seem unrelated?

Comment (diox) [reply src/olympia/scanners/tasks.py]: Yeah, it's just additional logging that I slipped in because you can never have enough logging :)
